src/Time/Bool.hs

```diff
@@ -0,0 +1,39 @@
+{-# LANGUAGE AllowAmbiguousTypes        #-}
+{-# LANGUAGE ConstraintKinds            #-}
+{-# LANGUAGE CPP                        #-}
+{-# LANGUAGE DataKinds                  #-}
+{-# LANGUAGE DeriveGeneric              #-}
+{-# LANGUAGE ExplicitForAll             #-}
+{-# LANGUAGE FlexibleContexts           #-}
+{-# LANGUAGE InstanceSigs               #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE ScopedTypeVariables        #-}
+{-# LANGUAGE TypeApplications           #-}
+{-# LANGUAGE TypeFamilies               #-}
+{-# LANGUAGE TypeInType                 #-}
+{-# LANGUAGE TypeOperators              #-}
+{-# LANGUAGE UndecidableInstances       #-}
+
+module Time.Bool
+       ( And
+       , If
+       , type (==%)
+       ) where
+
+import Data.Type.Equality (type (==))
+
+import Time.Rational (type (:%), Rat)
+
+-- | Type-level 'if'.
+type family If (x :: Bool) a b where
+    If 'True  a _ = a
+    If 'False _ b = b
+
+-- | Type-level 'and'.
+type family And (a :: Bool) (b :: Bool) where
+  And 'False _ = 'False
+  And _      x = x
+
+-- | A type family to compute equality of 'Rat's.
+type family (==%) (m :: Rat) (n :: Rat) where
```

Comment:
I discovered recently that operators can be written in infix form in definition, like this:

```haskell
type family (m :: Rat) ==% (n :: Rat) :: Rat where
```

(didn't try to compile this, so maybe it doesn't work actually).

Also, if you feel brave enough, you can try to make `==` operator overloaded just like @int-index did it for `*` and `/`. But I'm not sure it's possible with `==` because we need slightly different thing...

Reply:
There is already an overloaded type family for this is base:

https://hackage.haskell.org/package/base-4.10.1.0/docs/Data-Type-Equality.html#t:-61--61-

We need something like this to do overloading:

```
type instance m == n = m ==% n
```

Here we rely on the fact that `==%` has a fixed kind for the inputs (it is `Rat`), so the compiler will do kind-based dispatch.

Reply:
@vrom911 tried something like this. Turned out that `==` is closed type family. The error is the following:

```haskell
Illegal instance for closed family ‘==’
• In the type instance declaration for ‘==’
    |
148 | type instance (a :% b) == (c :% d) = (a == c) And (b == d)
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Reply:
Not true, I just checked in GHCi, I can define instances of `(==)`.

Reply:
I see, they changed it in 8.4